@zakkak I'm going to revert the last change because passing all existing system properties on causes some odd errors:

```
2025-10-28T13:28:31.3194232Z [ERROR] No plugin found for prefix '13' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (/home/runner/.m2/repository), google-maven-central (https://maven-central.storage-download.googleapis.com/maven2/), central (https://repo.maven.apache.org/maven2)] -> [Help 1]
2025-10-28T13:28:31.3197659Z org.apache.maven.plugin.prefix.NoPluginFoundForPrefixException: No plugin found for prefix '13' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (/home/runner/.m2/repository), google-maven-central (https://maven-central.storage-download.googleapis.com/maven2/), central (https://repo.maven.apache.org/maven2)]
2025-10-28T13:28:31.3200020Z     at org.apache.maven.plugin.prefix.internal.DefaultPluginPrefixResolver.resolve (DefaultPluginPrefixResolver.java:90)
2025-10-28T13:28:31.3201101Z     at org.apache.maven.lifecycle.internal.MojoDescriptorCreator.findPluginForPrefix (MojoDescriptorCreator.java:247)
2025-10-28T13:28:31.3202069Z     at org.apache.maven.lifecycle.internal.MojoDescriptorCreator.getMojoDescriptor (MojoDescriptorCreator.java:182)
2025-10-28T13:28:31.3203285Z     at org.apache.maven.lifecycle.internal.DefaultLifecycleTaskSegmentCalculator.calculateTaskSegments (DefaultLifecycleTaskSegmentCalculator.java:94)
2025-10-28T13:28:31.3204557Z     at org.apache.maven.lifecycle.internal.DefaultLifecycleTaskSegmentCalculator.calculateTaskSegments (DefaultLifecycleTaskSegmentCalculator.java:76)
2025-10-28T13:28:31.3206124Z     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:84)
2025-10-28T13:28:31.3206757Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
2025-10-28T13:28:31.3207250Z     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
2025-10-28T13:28:31.3207768Z     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
2025-10-28T13:28:31.3208260Z     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
2025-10-28T13:28:31.3208750Z     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
2025-10-28T13:28:31.3209205Z     at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
2025-10-28T13:28:31.3209727Z     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
2025-10-28T13:28:31.3210396Z     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
2025-10-28T13:28:31.3211161Z     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
2025-10-28T13:28:31.3211784Z     at java.lang.reflect.Method.invoke (Method.java:569)
2025-10-28T13:28:31.3212405Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
2025-10-28T13:28:31.3213114Z     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
2025-10-28T13:28:31.3213883Z     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
2025-10-28T13:28:31.3214550Z     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
```

I will add the 2 properties you indicated. If more are needed later on they can be added. What needs to be forwarded on is also dependant on what is relevant for specific tests.

--- a/integration-tests/maven/src/test/java/io/quarkus/maven/it/NativeAgentIT.java
+++ b/integration-tests/maven/src/test/java/io/quarkus/maven/it/NativeAgentIT.java
@@ -27,7 +27,8 @@ public class NativeAgentIT extends MojoTestBase {
         assertThat(runJvmITsWithAgent.getProcess().waitFor()).isZero();
 
         MavenProcessInvocationResult runNativeITs = running
-                .execute(List.of("verify", "-Dnative", "-Dquarkus.native.agent-configuration-apply"), Map.of());
+                .execute(TestUtils.nativeArguments("verify", "-Dnative", "-Dquarkus.native.agent-configuration-apply"),
+                        Map.of());
         assertThat(runNativeITs.getProcess().waitFor()).isZero();
     }
 }

--- a/integration-tests/maven/src/test/java/io/quarkus/maven/it/NativeImageIT.java
+++ b/integration-tests/maven/src/test/java/io/quarkus/maven/it/NativeImageIT.java
@@ -38,8 +38,9 @@ public class NativeImageIT extends MojoTestBase {
         final RunningInvoker running = new RunningInvoker(testDir, false);
 
         // trigger mvn package -Dnative -Dquarkus.ssl.native=true
-        final String[] mvnArgs = new String[] { "package", "-DskipTests", "-Dnative", "-Dquarkus.ssl.native=true" };
-        final MavenProcessInvocationResult result = running.execute(Arrays.asList(mvnArgs), Collections.emptyMap());
+        final List<String> mvnArgs = TestUtils.nativeArguments("package", "-DskipTests", "-Dnative",
+                "-Dquarkus.ssl.native=true");
+        final MavenProcessInvocationResult result = running.execute(mvnArgs, Collections.emptyMap());
         await().atMost(10, TimeUnit.MINUTES).until(() -> result.getProcess() != null && !result.getProcess().isAlive());
         final String processLog = running.log();
         try {

--- a/integration-tests/maven/src/test/java/io/quarkus/maven/it/TestUtils.java
+++ b/integration-tests/maven/src/test/java/io/quarkus/maven/it/TestUtils.java
@@ -1,5 +1,9 @@
 package io.quarkus.maven.it;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 import io.smallrye.common.os.OS;
 
 final class TestUtils {
@@ -11,5 +15,25 @@ final class TestUtils {
 
     static long getDefaultTimeout() {
         return DEFAULT_TIMEOUT;
+    }
+
+    static List<String> nativeArguments(String... initialArguments) {
+        final List<String> result = new ArrayList<>(Arrays.asList(initialArguments));
+        appendArgumentIfSet("quarkus.native.container-build", result);
+        appendArgumentIfSet("quarkus.native.builder-image", result);
+        appendArgumentIfSet("quarkus.native.container-runtime", result);
+        appendArgumentIfSet("quarkus.native.container-runtime-options", result);
+        return result;
+    }
+
+    private static void appendArgumentIfSet(String property, List<String> result) {
+        final String value = System.getProperty(property);
+        if (value != null) {
+            if (value.isEmpty()) {
+                result.add("-D" + property);
+            } else {
+                result.add(String.format("-D%s=%s", property, value));
+            }
+        }
     }
 }

